sbi/simulators/simutils.py

```diff
@@ -346,13 +377,56 @@ def __init__(
         self.return_type = return_type
 
     def log_prob(self, value) -> Tensor:
-        return torch.as_tensor(self.prior_scipy.logpdf(x=value), dtype=torch.float32)
+        return torch.as_tensor(self.prior_scipy.logpdf(x=value), dtype=self.return_type)
 
     def sample(self, sample_shape=torch.Size()) -> Tensor:
         return torch.as_tensor(
-            self.prior_scipy.rvs(size=sample_shape), dtype=torch.float32
+            self.prior_scipy.rvs(size=sample_shape), dtype=self.return_type
+        )
+
+    @property
+    def mean(self):
+        return self.prior_scipy.mean()
+
+    @property
+    def variance(self):
+        return self.prior_scipy.var()
+
+
+class PytorchReturnTypeWrapper(Distribution):
+    """Wrap PyTorch Distribution to return a given return type."""
+
+    def __init__(
+        self,
+        prior: Distribution,
+        return_type: Optional[torch.dtype] = torch.float32,
+        batch_shape=torch.Size(),
+        event_shape=torch.Size(),
+        validate_args=None,
+    ):
+        super().__init__(
+            batch_shape=batch_shape,
+            event_shape=event_shape,
+            validate_args=validate_args,
         )
 
+        self.prior = prior
+        self.return_type = return_type
+
+    def log_prob(self, value) -> Tensor:
+        return torch.as_tensor(self.prior.log_prob(value), dtype=self.return_type)
```

Comment:
I now see why you mentioned the `as_tensor` gradient tracking problem as a relevant issue.